MPCAutofill/cardpicker/tags.py

```diff
@@ -1,31 +1,59 @@
 """
 Tags for cards.
 """
-from __future__ import annotations
 
+import functools
 import re
+from typing import Optional
 
-from aenum import MultiValueEnum
+from aenum import MultiValueEnum, extend_enum
+
+from cardpicker import models
 
 
 class Tag(str, MultiValueEnum):
-    ALT_ART = "ALT ART", "ALTART", "ALT"
-    EXTENDED = "EXTENDED"
-    FULL_ART = "FULL ART", "FULLART", "FULL"
-    NSFW = "NSFW"
+    ALT_ART = "Alt Art", "Alternative Art", "Alternate Art", "Alt"
+    EXTENDED = "Extended", "EXTENDED Art"
+    FULL_ART = "Full Art", "FULL ART", "Fullart", "Full"
+    NSFW = "NSFW", "Nsfw"
 
     @staticmethod
-    def list_from_card_name(card_name: str) -> list[Tag]:
+    def extract_name_and_tags(name: Optional[str]) -> tuple[str, set[str]]:
+        """
+        This function unpacks a folder or image name which contains a name component and some number of tags
+        into its constituents.
+        Tags are wrapped in either [square brackets] or (parentheses), and any combination of [] and () can be used
+        within a single name.
+        """
+
+        if not name:
+            return "", set()
+
         tag_set = set()  # Use set to not have duplicate tags
-        pattern = r"\(([^\(\)]+)\)|\[([^\[\]]+)\]"  # Get content of () and []
-        tag_parts = re.findall(pattern, card_name)
+        tag_parts = re.findall(r"\(([^\(\)]+)\)|\[([^\[\]]+)\]", name)  # Get content of () and []
         cleaned_parts = list(map(lambda x: x[0] if len(x[0]) != 0 else x[1], tag_parts))
+        name_with_no_tags = functools.reduce(
+            lambda mutated_name, tag: mutated_name.replace(f"[{tag}]", "").replace(f"({tag})", "").strip(),
+            [name, *cleaned_parts],
+        )
         for tag_part in cleaned_parts:
             raw_tags = tag_part.split(",")
             for raw_tag in raw_tags:
                 try:
-                    tag_set.add(Tag(raw_tag.strip().upper()))
+                    tag_set.add(Tag(raw_tag.strip().title()))
```

Comment:
Why do you use title instead of upper ? For display in the front-end ?

Reply:
correct, i though titlecase was more appropriate for display in the frontend